### ISSUE TITLE
Push notifications on iOS broke for me between 10.4.0 and 10.5.0

### ISSUE THREAD

**Issue by lorenzk** [EddyVerbruggen/nativescript-plugin-firebase#1589]
Hi @EddyVerbruggen and thanks for creating such a fundamental part of our news app! Our app has developed problems on iOS after upgrading from 10.4 to 10.5 and I cannot figure out why - maybe you or someone else in the community can help?

**In version 10.4.0**
Notifications arrive both in the foreground and background as they should, tapping them opens a detail view in the app via the `onMessageReceivedCallback` .

**In versions 10.5.0, 10.5.1, and 10.5.2**
Notifications do not arrive when the app is in the foreground. They do arrive when I put the app in the background, but tapping the notification only opens the app, without calling the `onMessageReceivedCallback`. After force-closing the app, notifications still do not trigger anything in the foreground, but do arrive in the background and tapping them runs the callback.

I confirmed this both on TestFlight and debugging via cable connected to my laptop.

**Settings and version numbers:**
- `showNotifications: true`: Setting this to `false` for some reason breaks notifications on first run. Things only work after force-closing the app at least once.
- `showNotificationsWhenInForeground: false`
- `nativescript@6.5.0` as well as everything else at `latest`, except the version of `nativescript-plugin-firebase` in each of my test runs.

I tried looking at the changes between 10.4.0 and 10.5.0 and inserting `console.log` calls to maybe find the difference in execution paths but could not see anything out of the ordinary.

Just in the off chance that this is relevant: we use AWS Pinpoint to trigger messages via APNS.

Thank you for your time!

**Comment by manojdcoder** [comment]
If you revert back to 10.4.0, everything still works as expected?

**Comment by lorenzk** [comment]
Yes, exactly. I went back and forth between versions multiple times to pinpoint the issue.

**Comment by allangoncalves** [comment]
@lorenzk were you able to find the source of the problem?

**Comment by lorenzk** [comment]
@allangoncalves No, we are keeping the downgrade for now :-(

**Comment by dbenninger** [comment]
I have the same issue upgrading from 9.0.2 to 10.6.3. I get the push token, but I just don't get any push messages anymore on iOS. 
I upgraded to only 10.4.0 and that worked. I also upgraded to 10.5.2 and that worked too. When I upgraded to 10.6.0, it stopped working again. Was not able to figure out why yet...
